internal/tfdiags/diagnostic_base.go

```diff
@@ -38,3 +38,25 @@ func (d diagnosticBase) FromExpr() *FromExpr {
 func (d diagnosticBase) ExtraInfo() interface{} {
 	return nil
 }
+
+func (d diagnosticBase) Equals(otherDiag ComparableDiagnostic) bool {
```

Comment:
since `diagnosticBase` can't implement reliable equality because it doesn't contain the location data, we should probably remove this method

Reply:
That would mean sourceless diagnostics become incomparable - what do you suggest we do in tests that need to test for equality then? https://github.com/hashicorp/terraform/blob/43ee22f63811c86d5210000368093e5544b22a2a/internal/tfdiags/sourceless.go#L10-L16

Reply:
On a second look I don't see any test failures anymore - not sure where I saw them before. Either way, I removed the comparison method.

Reply:
I don't know why I cannot reproduce those failures locally but the CI does still show 4 failing tests.

We can implement a separate comparison with `cmp.Diff` for those tests I suppose - does that sound like a sensible way forward?

Reply:
`TestBackendConfig_Proxy` currently uses `cmp.Diff` directly, does anything change if `tfdiags.AssertDiagnosticsMatch` is used instead?